source/shared/core_conn.cpp

```diff
@@ -150,34 +150,134 @@ sqlsrv_conn* core_sqlsrv_connect( _In_ sqlsrv_context& henv_cp, _In_ sqlsrv_cont
     
     build_connection_string_and_set_conn_attr( conn, server, uid, pwd, options_ht, valid_conn_opts, driver, conn_str TSRMLS_CC );
     
-    bool is_missing_driver = false;
+#ifndef _WIN32
+    if( conn->ce_option.enabled ) {
+        // when AE is enabled, must use ODBC driver 17
```

Comment:
feature is Always Encrypted, option is CE (your comment in the close brace correctly refers to this